crates/fluvio-cluster/src/lib.rs

```diff
@@ -137,4 +143,89 @@ mod common {
         };
         Ok(cert_paths)
     }
+
+    #[derive(Debug, Default, Clone, PartialEq, Eq, Serialize, Deserialize)]
+    #[serde(rename_all = "snake_case")]
+    pub enum InstallationType {
+        #[default]
+        K8,
+        Local,
+        LocalK8,
+        ReadOnly,
+    }
+
+    impl FromStr for InstallationType {
+        type Err = String;
+
+        fn from_str(s: &str) -> Result<Self, Self::Err> {
+            if s.eq_ignore_ascii_case("k8") {
+                return Ok(Self::K8);
+            }
+            if s.eq_ignore_ascii_case("local") {
+                return Ok(Self::Local);
+            }
+            if s.eq_ignore_ascii_case("local-k8") {
+                return Ok(Self::LocalK8);
+            }
+            if s.eq_ignore_ascii_case("local_k8") {
+                return Ok(Self::LocalK8);
+            }
+            if s.eq_ignore_ascii_case("localk8") {
+                return Ok(Self::LocalK8);
+            }
+            if s.eq_ignore_ascii_case("read_only") {
+                return Ok(Self::ReadOnly);
+            }
+            if s.eq_ignore_ascii_case("read-only") {
+                return Ok(Self::ReadOnly);
+            }
+            if s.eq_ignore_ascii_case("readonly") {
+                return Ok(Self::ReadOnly);
+            }
```

Comment:
Should  these different spelling variations be allowed in the installation type parsing from str?

Reply:
This is involved in the only place where we pass the installation type as an input:
```
fluvio profile add <PROFILE_NAME> <CLUSTER_ADDRESS> [INSTALLATION_TYPE] 
```
and while using it I realized that I didn't want to remember if it should be `local-k8` or `local_k8` etc. So, I added support for all of them.

However, in start command:
```
fluvio cluster start --local-k8
```
it is a CLI argument with --, so it only supports one form.